pkg/querier/querier.go

```diff
@@ -335,6 +335,12 @@ func (q querier) Select(sortSeries bool, sp *storage.SelectHints, matchers ...*l
 	sp.Start = startMs
 	sp.End = endMs
 
+	// For series queries without specifying the start time, we prefer to
+	// only query ingesters and not to query maxQueryLength to avoid OOM kill.
+	if sp.Func == "series" && startMs == 0 {
```

Comment:
I think it's better if we merge this to line 307 above.

Reply:
I see. I wanted to merge the two but to do this we need to do L318 ~ L336 first before L307. 
This is probably a behavior change because we will do the time range validation first and then send to the metadata querier.

Reply:
I was thinking we can do something along the line

```
userID, err := tenant.TenantID(ctx)
if err != nil {
  return storage.ErrSeriesSet(err)
}
startMs, endMs, err := validateQueryTimeRange(ctx, userID, sp.Start, sp.End, q.limits, q.maxQueryIntoFuture)

if sp == nil {
  sp = &storage.SelectHints{Start: q.mint, End: q.maxt}
} else if sp.Func == "series" && (!q.queryStoreForLabels ||  startMs == 0) {	
   return q.metadataQuerier.Select(true, sp, matchers...)
} 

if err == errEmptyTimeRange {
    return storage.NoopSeriesSet()
} else if err != nil {
   return storage.ErrSeriesSet(err)
}
```

Essentially if start is `0` we treat it as if `queryStoreForLabels` is `false`. 

There is an edge case where user actually specifies "0" for "start", and Cortex would still return last 24 hours data, but I think this behaviour is ok for the edge case.

Reply:
OK, I think this could work...
But is it worth? For readability I prefer to separate them. It is just 1 duplicated line.
This is not very clear tbh. 
`if sp.Func == "series" && (!q.queryStoreForLabels ||  (err != nil && startMs == 0))`

Reply:
Yea that's why I was thinking maybe `validateQueryTimeRange` should return `-1`s when `err` is not nil. But this is fine, I don't think we need to do it change anything for now; it's up to you. 


I will let @alanprot have to final approval; I think this PR is ok.